src/redux/weather.js

```diff
@@ -0,0 +1,43 @@
+import { createAction, createAsyncThunk, createReducer } from '@reduxjs/toolkit';
+
+const FETCH_WEATHER = 'FETCH_WEATHER';
+const PICK_CITY = 'PICK_CITY';
+
+const cities = ['Paris', 'Barcelona', 'Copenhagen', 'Vienna', 'Berlin', 'Lisbon', 'Madrid', 'New York', 'Buenos Aires', 'Montevideo', 'Kampala', 'Budapest', ' Beijing', 'Tripoli', 'Kurkata'];
+
+const fetchoneCityWeather = async (city) => {
+  const resp = await fetch(`https://weather.visualcrossing.com/VisualCrossingWebServices/rest/services/timeline/${city}/2023-02-17?unitGroup=metric&include=days&key=9XBYDAN3EHGMRUXXB7VHBZYFV&contentType=json`);
```

Comment:
[OPTIONAL]

- I will strongly recommend you don't expose your API key which is a secret to prevent a malicious attack on your website. Kindly ensure you always save your secret keys to the environment variable (.env) so that it won't be pushed to git. 👍🏽 You can [read](https://create-react-app.dev/docs/adding-custom-environment-variables/) more on how to use environment variable in react